### ISSUE TITLE
upgrade react-scripts to latest version 5.0.1

### ISSUE THREAD

**Issue by erasta** [argosp/trialdash#331]
currently react-scripts version is 2.x
when upgrading react-scripts several packages and components break, graphql included.
please upgrade react-scripts so we can advance to modern maps and material ui.

**Comment by erasta** [comment]
@shimlinnovate @yehudarav any update on this?

**Comment by alex-bondarev-linnovate** [comment]
There's been a lot of discussion about the future of Create React App recently. There's a ~[2-month-old commend by Dan](https://github.com/reactjs/react.dev/pull/5487#issuecomment-1409720741) (part of the React team) about the future of CRA and there's a [very recent Twitter thread](https://twitter.com/dan_abramov/status/1636827365677383700) too.
TLDR of those links is - _CRA will be changed into something else_.

So, to complete this task we'll need to try and predict the future of this project, let me highlight possible routes:

**1. This app will need to be changed from client-side-only to be hybrid** (Server-Side Rendering + client interactivity)

If we see the potential of some parts of the app to be prerendered on the server and produce HTML for a faster initial feedback to the user - we'd need to migrate the app to NextJS or Remix.

**2. This app will remain client-side only**

If we don't see a need for SSR, we'll have **2 options**, each has pluses and minuses:

**Option 1.** Do what the task is about - update react-scripts to v5.

_Drawbacks:_

- _seems to break old apollo client_

As a result - apollo client will need to be updated too. Unfortunately it currently conflicts with Material UI, npm cannot create dependency tree if we update react-scripts, apollo client, but not Material. It seems that all 3 parts will need to be updated simultaneously - react-scripts, apollo client and material UI, which complicates things.

- _updating react-scripts is not future-proof_

One of the last updates in its current form, the future of `react-scripts` is super unclear, it's highly likely that the project will be discontinued/changed/transformed/abandoned, so we'll still need to migrate from or adapt it anyways when such changes occur. See links above for more info.

_Benefits:_

I see no benefits. The update is not easy and has breaking changes.

**Option 2.** Migrate to alternative tech - [Vite](https://main.vitejs.dev/guide/)

_Drawbacks:_

- _will require components change_, like renaming of files from .js to .jsx
- _absolute paths will need to be handled a bit differently._ Vite provides configuration for it.
- _all components will need to be rechecked in both versions - dev and production_, because different mechanisms are used for Dev and Build (can provide more details if needed)

_Benefits:_

- _officially recommended_ as a frameworkless approach in the React [documentation](https://react.dev/learn/start-a-new-react-project)
- _relatively easy migration of most of the codebase_
- _future-proof, on-going project with an active community_

To recap, my recommendations would be to:

a) decide if we need SSR (if yes then migrate to NextJS),
b) if no, then migrate to Vite as a future-proof alternative of react-scripts.

I need you decision on this.

**Comment by yehudarav** [comment]
Hi.
How long will option 1 and 2.2 will take?

**Comment by shimlinnovate** [comment]
@yehudarav Hi Yehuda please read @alex-bondarev-linnovate Research regarding @erasta  upgrade request, we need to decide on a path to continue the process

**Comment by yehudarav** [comment]
@Eran Geva ***@***.***>  What do you think?

**Comment by erasta** [comment]
Hi @alex-bondarev-linnovate thank you for a great review of the different directions our react app could take.

I believe that a revolutionary change to SSR like NextJs might be problematic here, as it will probably need a full rewrite and take a long time to get a minimal system working. According to @yehudarav the next phase of experiments is needed soon. 

So staying with an evolutionary upgrade of the client side react app seems like the right way. As I see it Apollo and MaterialUI needs to be upgraded anyways, either with react-scripts or with Vite.

If you managed to recreate the application with Vite and see the correct experiments and trials, a.k.a the Graphql DB connection working and all, this sounds like a great advancement!
The map not working in that configuration is expected, as we are currently using an old react-leaflet package, which we are forced to because of react-script (upgrading it is the incentive of this move).

Please publish this branch with the operable tables views, and i'll branch from it to tweak the map into working.

Many thanks!

**Comment by alex-bondarev-linnovate** [comment]
@erasta can you help with the following?

```
ERROR: Permission to argosp/argos.git denied to alex-bondarev-linnovate.
fatal: Could not read from remote repository.
```

**Comment by erasta** [comment]
@alex-bondarev-linnovate i think this works only in https, at least for me, try to clone like in https://github.com/argosp/argos#readme

**Comment by erasta** [comment]
Did you manage to **read** from the repo but not to **write**?
There might be a permissions problem.

**Comment by erasta** [comment]
Cloning should work for you but since it doesn't there might be some permissions missing.
I do not have permissions to give you permissions. maybe @shimlinnovate?

Be aware that there are `trialdash/Dockerfile` and `utils/dockers/trialdash/Dockerfile` files, one for prod and one for dev, so you'll need to change also the `argosp` repo.

**Comment by alex-bondarev-linnovate** [comment]
Was I correct in creating a feature branch in the submodule or should I have created a git branch in the main (`argos`) repo?

**Comment by alex-bondarev-linnovate** [comment]
Sorry, the problem was with Password authentication. We'll need to reconfigure submodules to not pull via https. Once I switched the remote url from https to Git, it allowed me to push to a branch - here it is https://github.com/argosp/trialdash/tree/AB/vite

**Comment by alex-bondarev-linnovate** [comment]
Alas, I overwrote the master changes so it seems ( this branch reapplies the changes made after I pulled it, I'll need to recreate the changes and apply them onto latest `master`. I was a bit naive to think that I can just copy `src` from my `vite` one. I'll need to redo the branch @erasta

**Comment by alex-bondarev-linnovate** [comment]
@erasta @yehudarav @shimlinnovate I've just repushed all of the changes that need to be made when migrating to Vite, it's 1 commit ahead of the latest master, so contains all of your work.

You can browse this commit and see what changes I made (they are not quite big).

![image](https://user-images.githubusercontent.com/127842115/227124135-2f1d9f81-1d11-473b-be2e-80e3bf63c1d1.png)

Re: leaflet update. Unfortunately, the components need to be changed if you wish to update to newest leaflet. Upon changing the version locally it installs (no npm conflicts), but requires component updates, which is expected I guess. I'm seeing the following in the console.

```
does not provide an export named 'Map' (MapWithImage.jsx)
```

**Comment by yehudarav** [comment]
So do you have an estimate of the time it takes to perform these updates?

**Comment by erasta** [comment]
@alex-bondarev-linnovate Looks good!
Everything is working ok on the dev environment correctly, including the hot reload.
I will continue the upgrade of react-leaflet if you say there are not NPM conflicts.
I'm wondering if the production build of trialdash, that is invoked by CircleCi when merging to master, will also work correctly.
Is there an easy way to check that before merging to master?
If the merge will go ok, this bug would be closed.

**Comment by alex-bondarev-linnovate** [comment]
@erasta RE trialdash update. Let me consult my crystal ball - yes, the merge will be without errors :smile: Kidding. Vite provides a way to test it. In `package.json` you can see the `preview` script. So `vite preview` will probably try and build the production build temporarily and maybe open it on the same port, maybe different port, don't know for sure. The `build` command should also be taken a look at, simply because it uses the `env-cmd .env` part which may be a requirement for react-script but may not be one for Vite. My recommendation will be to start from there and see how these commands work locally.

**Comment by erasta** [comment]
@alex-bondarev-linnovate crystal ball is a great method 🤣
i am trying to change https://github.com/argosp/trialdash/blob/master/.circleci/config.yml#L45 to your branch, will see how that works on prod directly. will update, thanks.

**Comment by alex-bondarev-linnovate** [comment]
OMG, I'd first test locally, just in case @erasta Why straight into production? Sounds so risky to me

**Comment by erasta** [comment]
Noone actually uses the production at the moment, that is only mock data, they clone and build it on another machine.

**Comment by alex-bondarev-linnovate** [comment]
I'd go for changing the docker run commands. By default it runs the `dev` command, I'd find that place and either a) change it temporarily to 'preview' and see how/if it build, or b) make it a variable, so that local env could start either as development, or as preview. Maybe put it in the .env or something. Just from the top of my head though.

**Comment by erasta** [comment]
changed the command on https://github.com/argosp/argos/blob/master/docker-compose.yml#L41 to `yarn run build && yarn run serve`
changed the serve command `https://github.com/argosp/trialdash/blob/AB/vite/package.json#L53` to `serve -s dist` as it builds to dist
and now all is working locally.

not sure that on production the circleci actually does that, do you know? maybe can you have allocated time to join a screenshare and test this? @alex-bondarev-linnovate
FYI @shimlinnovate @yehudarav

**Comment by alex-bondarev-linnovate** [comment]
@erasta Well done! RE Circle CI - I'm afraid I have little information on the prod environment or on CI, I'm a relatively new guy here. In my opinion - it should be like you describe, we use dev for local development (hot reloading, etc.) and use build for prod (minified and optimised assets, etc.). If it's not set up this way - we'll probably need to find out why, maybe there are reasons. If there are no - I'd go for double-checking and changing the CI config to include the build command
